hazelcast-client/src/main/java/com/hazelcast/client/config/ClientConfigXmlGenerator.java

```diff
@@ -497,13 +498,13 @@ private static void ssl(XmlGenerator gen, SSLConfig ssl) {
            .close();
     }
 
-    private static void aws(XmlGenerator gen, ClientAwsConfig aws) {
+    private static void aws(XmlGenerator gen, AwsConfig aws) {
```

Comment:
https://github.com/hazelcast/hazelcast/pull/15012#issuecomment-494274667
Did you see my comment here? When I add a test to ClientConfigXmlGeneratorTest as follows I can see that it fails:
```
@Test
    public void testKubernetes() {
        KubernetesConfig kubernetesConfig = clientConfig.getNetworkConfig().getKubernetesConfig();
        kubernetesConfig.setEnabled(true);

        KubernetesConfig kubernetesConfig1 = newConfigViaGenerator().getNetworkConfig().getKubernetesConfig();
        assertTrue(kubernetesConfig1.isEnabled());
    }
```
So, I think aws support is there but others are missing. They needs to be added. Hence I open the issue. Does that make sense ? @leszko

Reply:
@sancar thanks for clarifying.

1. You're right that the support for other strategies (`kubernetes`, `gcp`, etc.) is missing. Please open an issue.

2. However, even though AWS support is there, you can remove it, because it's deprecated. It's parameters are not up to date (you can compare to the [current AWS Discovery Strategy Parameters](https://github.com/hazelcast/hazelcast-aws#java-based-configuration)). We should remove it and add the generic solution like it's done [here](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java#L709).